dsgrid/config/dimensions.py

```diff
@@ -694,3 +682,35 @@ def create_project_dimension_model(model, is_base):
     data = create_dimension_common_model(model).dict()
     data["is_base"] = is_base
     return ProjectDimensionModel(**data)
+
+
+def check_display_name(display_name: str):
+    """Check that a dimension display name meets all requirements.
+
+    Raises
+    ------
+    ValueError
+        Raised if the display_name is invalid.
+    """
+    if display_name == "":
+        raise ValueError("display_name cannot be empty")
+    if REGEX_VALID_REGISTRY_DISPLAY_NAME.search(display_name) is None:
+        raise ValueError(f"{display_name=} does not meet the requirements")
+    return display_name
+
+
+def generate_dimension_query_name(dimension_query_name: str, display_name: str) -> str:
+    """Generate a dimension query name from a display name.
+
+    Raises
+    ------
+    ValueError
+        Raised if the dimension_query_name was set by the user and does not match the generated
+        name.
+    """
+    generated_query_name = display_name.lower().replace(" ", "_").replace("-", "_")
+
+    if dimension_query_name is not None and dimension_query_name != generated_query_name:
+        raise ValueError(f"dimension_query_name cannot be set by the user: {dimension_query_name}")
```

Comment:
If dimension query name cannot be auto-generated, doesn't that mean user has to set it instead?

Reply:
This is refactored code that maintains existing behavior. It will only fail if the user sets the name to a string that does not match the auto-generated string.